### ISSUE TITLE
id entries in namelist_definition must be lowercase

### ISSUE THREAD

**Issue by jedwards4b** [ESMCI/cime#3144]
The id field in namelist definition files must be lower case.  Use of upper case or mixed case strings in this field will cause errors.   This was noticed in regard to changing a namelist variable via user_nl_* files.

**Comment by ekluzek** [comment]
Since, FORTRAN doesn't distinguish the case of variables in a namelist, I'm not sure this is completely a bad thing. But, good for us to watch out for...

**Comment by billsacks** [comment]
@jedwards4b should this be closed as a wontfix (based on what you said in person a little while ago)? If we keep it open, what do you think about changing the title to, "entries in namelist_definition must be lowercase"?

**Comment by brian-eaton** [comment]
For those of us who weren't in the hallway, why is this hard to fix in cime?

**Comment by jedwards4b** [comment]
It's because variables read from user_nl_*** are downcased on input and so they will only match downcase variables from the namelist_definition file.   I tried removing the downcase from the user_nl_*** read and that lead to a bunch of issues, and I tried downcasing the namelist_definition variables on read and that also lead to a host of new problems.  This seems to be the best approach until someone has resources to fix the problem properly.
